File: rwebspec/pages/payment_page.rb
require File.join(File.dirname(__FILE__), "abstract_page.rb")

class PaymentPage < AbstractPage

  def initialize(driver)
    super(driver, "") # <= TEXT UNIQUE TO THIS PAGE
  end

def enter_holder_name(holder_name)
  enter_text("holder_name", holder_name)
end

def enter_card_number(card_number)
  enter_text("card_number", card_number)
end

def enter_expiry_month(expiry_month)
  select_option("expiry_month", expiry_month)
end

def enter_expiry_year(expiry_year)
  select_option("expiry_year", expiry_year)
end

def click_pay_now
  click_button("Pay now")
end

def select_card_type(card_type)
  click_radio_option("card_type", card_type)
end

end

File: performance-comparision/pages/payment_page.js
const AbstractPage = require('./abstract_page');

var webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until;

class PaymentPage extends AbstractPage {

  constructor(driver) {
    super(driver);
  }

  async enterHolderName(holder_name) {
    await this.driver.locator("[name=holder_name]").fill(holder_name);
  }

  async enterCardNumber(card_number) {
    await this.driver.locator("[name=card_number]").fill(card_number);
  }

  async enterExpiryMonth(expiry_month) {
    await this.driver.selectOption("select[name='expiry_month']", expiry_month);
  }

  async enterExpiryYear(expiry_year) {
    await this.driver.selectOption("select[name='expiry_year']", expiry_year);
  }

  async selectCardType(card_type) {
     await this.driver.locator("//input[@name='card_type' and @value='" + card_type +  "']").click();
   //  await driver.find_element(:xpath, "//input[@name='card_type' and @value='" + card_type +  "']").click
  }

  async clickPayNow() {
    await this.driver.click("input:has-text('Pay now')");
  }

};

module.exports = PaymentPage;
